Automattic/newspack-blocks
# Pop up block popup backend

### All Submissions:






### Changes proposed in this Pull Request:



Related https://github.com/Automattic/newspack-issues/issues/11

This PR does most of the work for implementing an AMP-compatible pop-ups solution. The frontend/admin styling can be fine-tuned in a follow-up PR; this PR is mostly interested in the logic for creating and displaying the pop-up.

### How to test the changes in this Pull Request:

0. Make sure you're running the latest version of the Gutenberg plugin.
1. In the Customizer  > Widget Blocks section, insert a Pop Up block into one of the widget areas (Sidebar, Above/Below Content, or any sidebars created by Super Cool Ad Inserter):
<img width="301" alt="Screen Shot 2019-09-18 at 10 47 22 AM" src="https://user-images.githubusercontent.com/7317227/65172482-dd0dc380-da01-11e9-9b17-0b79af880480.png">
2. In a new incognito window, visit a single post. You should get one popup. Visiting another single post should not show a popup. You should not get a pop-up when visiting a post when logged in or when visiting a non-post. Pop-ups should work correctly in AMP and non-AMP modes.

<img width="578" alt="Screen Shot 2019-09-18 at 10 56 51 AM" src="https://user-images.githubusercontent.com/7317227/65173122-0d099680-da03-11e9-994e-d0b49e5644bd.png">


### Questions:

- How should we do the AMP-CORS header handling for this? Is that something we need to take into account?
- Should we hide the pop up block from the general block editor? It doesn't necessarily make sense for people to use the block in their articles.
- Is there a better/clearer way than the red border in the block editor for making the pop-up block "show" that it's a pop-up?

### Other information:
